### PR TITLE
[apm-server]Fix fullnameOverride setting

### PR DESCRIPTION
**Pull request by kawat55**
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [ ] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

File: apm-server/tests/apmserver_test.py [hunk 1]
```diff
@@ -296,3 +296,21 @@ def test_priority_class_name():
...
     assert priority_class_name == "highest"
+
+
+def test_setting_fullnameOverride():
+    config = """
+fullnameOverride: "apm-server-custom"
+"""
+    r = helm_template(config)
+
+    custom_name = 'apm-server-custom'
```
Comment (jmlrt) [inline apm-server/tests/apmserver_test.py]:
```suggestion
    custom_name = "apm-server-custom"
```

**Comment by jmlrt** [comment]
jenkins test this please

File: apm-server/tests/apmserver_test.py [hunk 2]
```diff
@@ -296,3 +296,21 @@ def test_priority_class_name():
...
+    custom_name = "apm-server-custom"
+    assert custom_name in r["deployment"]
+    assert custom_name in r["service"]
+
+    assert (
+        r["deployment"][custom_name]["spec"]["template"]["spec"]["containers"][0][
+            "name"
+        ]
+        == project
+    )
```
Comment (jmlrt) [inline apm-server/tests/apmserver_test.py]: I think black is now complaining about the missing newline

Comment (kawat55) [reply apm-server/tests/apmserver_test.py]: Sorry, fixed.

**Comment by jmlrt** [comment]
jenkins test this please

**Pull request merged**